assets/payment/change_plan_script.js

```diff
@@ -13,10 +32,20 @@
 			.then((resp) => resp.json())
 			.then((response) => {
 				if (response.success) {
-					window.unonload = window.opener.redirectToCallback(
-						'notices-code=payment'
-					);
-					window.close();
+					if (
+						response.data &&
+						response.data.status === 'requires_action'
+					) {
+						confirmPayment(
+							response.data.clientSecret,
+							response.paymentMethod
+						);
+					} else {
+						window.unonload = window.opener.redirectToCallback(
+							'notices-code=payment'
+						);
+						window.close();
```

Comment:
This code is duplicated in line 13-16. Remove the duplication by creating a function